Documentation/GeneralConventions/GuidesXml.rst

```diff
@@ -21,117 +17,227 @@ Example:
     a :file:`Documentation/Settings.cfg` you can use an automatic migration
     tool to :ref:`migrate the settings.cfg into a guides.xml <migrate_guides_xml>
 
+..  contents:: Table of Contents:
+    :local:
+
+
 .. _settings-guides:
 
 The `<guides>` tag
 ==================
 
 It is used for general configuration during parsing and rendering.
 
+The following settings can be relevant for TYPO3 themed documentation:
+
+..  _settings-guides-default-code-language:
+
+default-code-language
+---------------------
+
+..  confval:: default-code-language
+    :name: guides-default-code-language
+    :type: string
+
+    Can be used to set the code-language that should be used for
+    :ref:`code-blocks <codeblock-directive>` if no language is set.
+
+    We advice to set the language to be used in each code-block however.
+
+..  _settings-guides-default-code-language-example:
+
+Example: Use `typoscript` as default langauge for code-blocks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+..  code-block:: xml
+    :caption: EXT:my_extension/Documentation/guides.xml (excerpt)
+
+    <?xml version="1.0" encoding="UTF-8"?>
+    <guides xmlns="https://www.phpdoc.org/guides" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd"
+            default-code-language="typoscript"
+    >
+        <!-- [...] -->
+    </guides>
+
+..  _settings-guides-max-menu-depth:
+
+max-menu-depth
+--------------
+
+..  confval:: max-menu-depth
+    :name: guides-max-menu-depth
+    :type: integer
+    :Default: `PHP_INT_MAX`
+
+    Limits the main menu to a certain amount of levels. As level creation is
+    one of the main bottle-necks that can slow down rendering it can be helpful
+    to the levels in the menu in some use cases.
+
+..  _settings-guides-max-menu-depth-example:
+
+Example: Limit the menu to 2 levels
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The changelog (documentation of the extension "core") contains several thousand
```

Comment:
The changelog (documentation of the extension "core") consists of several thousands of